power_grid_model_c/power_grid_model/include/power_grid_model/component/asym_line.hpp

```diff
@@ -0,0 +1,115 @@
+#pragma once
+
+#include <numeric>
+
+#include <iostream>
+#include "branch.hpp"
+
+#include "../auxiliary/input.hpp"
+#include "../auxiliary/output.hpp"
+#include "../auxiliary/update.hpp"
+#include "../calculation_parameters.hpp"
+#include "../common/common.hpp"
+#include "../common/three_phase_tensor.hpp"
+#include "../common/matrix_utils.hpp"
+#include "line_utils.hpp"
+
+namespace power_grid_model {
+
+class AsymLine : public Branch {
+  public:
+    using InputType = AsymLineInput;
+    using UpdateType = BranchUpdate;
+    static constexpr char const* name = "asym_line";
+
+    explicit AsymLine(AsymLineInput const& asym_line_input, double system_frequency, double u1, double u2)
+        : Branch{asym_line_input}, i_n_{asym_line_input.i_n}, base_i_{base_power_3p / u1 / sqrt3} {
+        if (cabs(u1 - u2) > numerical_tolerance) {
+            throw ConflictVoltage{id(), from_node(), to_node(), u1, u2};
+        }
+
+        ComplexTensor<asymmetric_t> c_matrix = compute_c_matrix_from_input(asym_line_input);
+        ComplexTensor<asymmetric_t> z_series = compute_z_series_from_input(asym_line_input);
+
+        y_series_abc_ = inv(z_series);
+        y_shunt_abc_ = 2 * pi * system_frequency * c_matrix * 1.0i;
```

Comment:
We forgot one thing. To translate the physical values into per-unit values. Without this it will not work in the mathematical core. Please have a look at line implementation. The `y_series_` and `y_shunt_` has to be divided by `y_base`.

https://github.com/PowerGridModel/power-grid-model/blob/8d410ae88f345e778d22a110893ab7aeb141124d/power_grid_model_c/power_grid_model/include/power_grid_model/component/line.hpp#L30-L34

Reply:
It's correct.